egs/aishell/s10/chain/model.py

```diff
@@ -174,6 +170,13 @@ def forward(self, x):
         # tdnnf requires input of shape [N, C, T]
         for i in range(len(self.tdnnfs)):
             x = self.tdnnfs[i](x)
+            # stride manually, do not stride context
+            if self.tdnnfs[i].time_stride == 0:
+                cur_context = sum(self.time_stride_list[i:])
+                x_left = x[:, :, :cur_context]
+                x_mid = x[:, :, cur_context:-cur_context:self.frame_subsampling_factor]
+                x_right = x[:, :, -cur_context:]
+                x = torch.cat([x_left, x_mid, x_right], dim=2)
```

Comment:
I'm surprised that you are doing this manually rather than using a 1d convolution.  This could be quite slow.

Reply:
I want subsample the length of window only rather than left_context and right_context. And this is slower than before training, but it worked. please help me to write this 1d convolution.

Reply:
What might have happened here is that you tripled the dimension in the middle of the network.
This would have led to a system with many more parameters for your "dilation" system.

Reply:
Just subsample the t_out_length from (24+150+24) to (24+50+24) manually, the number of parameters will not increase than **stride kernel(2,2)** version. I explained this code behaviour in the picture below.